### PR TITLE
[WIP][Parser][SR-1952] Added @escaping attribute parsing

### PR DESCRIPTION
**Pull request by tanadeau**
#### What's in this pull request?

Adds parsing of the new `@escaping` attribute.
#### Resolved bug number: WIP of ([SR-1952](https://bugs.swift.org/browse/SR-1952))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

File: include/swift/AST/Types.h [hunk 1]
```diff
@@ -2186,15 +2187,18 @@ class AnyFunctionType : public TypeBase {
...
       : ExtInfo(Rep, IsNoReturn, Throws) {
       Bits |= (IsAutoClosure ? AutoClosureMask : 0);
       Bits |= (IsNoEscape ? NoEscapeMask : 0);
+      Bits |= (IsExplicitlyEscaping ? ExplicitlyEscapingMask : 0);
     }
 
     bool isNoReturn() const { return Bits & NoReturnMask; }
     bool isAutoClosure() const { return Bits & AutoClosureMask; }
     bool isNoEscape() const { return Bits & NoEscapeMask; }
+    bool isExplicitlyEscaping() const { return Bits & ExplicitlyEscapingMask; }
```
Comment (tanadeau) [inline include/swift/AST/Types.h]: Note that everything involved with "explicitly escaping" in this PR is temporary and will be removed once the default is changed.

**Comment by CodaFi** [comment]
@swift-ci please test.

**Comment by tanadeau** [comment]
The checks passed. Can this be merged?

File: include/swift/AST/Types.h [hunk 2]
```diff
@@ -2159,11 +2159,12 @@ class AnyFunctionType : public TypeBase {
     //   |representation|isAutoClosure|noReturn|noEscape|throws|
     //   |    0 .. 3    |      4      |   5    |    6   |   7  |
```
Comment (CodaFi) [inline include/swift/AST/Types.h]: Nitpick: This doc comment needs to be updated.

**Comment by CodaFi** [comment]
Yep.  Good work!  👍

**Pull request merged**